src/Formatter/Formatter.php

```diff
@@ -238,16 +279,22 @@ public function onAfterOutlineTested(BehatEvent\AfterOutlineTested $event)
     {
         /** @var TestResults $testResults */
         $testResults = $event->getTestResult();
-        $stepCount = count($event->getOutline()->getSteps());
+        if ($this->currentFeature->getBackground()) {
+            $backgroundStepCount = count($this->currentFeature->getBackground()->getSteps());
+        }
+        else {
+            $backgroundStepCount = 0;
+        }
+        $scenarioStepCount = count($event->getOutline()->getSteps());
         foreach ($testResults as $i => $testResult) {
             $example = clone $this->currentScenario;
 
             // use correct line number of example row
             $line = $event->getOutline()->getExampleTable()->getRowLine($i + 1);
             $example->setLine($line);
-
             // remove all steps and attach only steps for that example row
-            $steps = array_slice($example->getSteps(), $i * $stepCount, $stepCount);
+            $stepOffset = $i * ($scenarioStepCount + $backgroundStepCount) + $backgroundStepCount;
```

Comment:
Bug. Should be:
$stepOffset = $i * ($scenarioStepCount + $backgroundStepCount);